cmd/conversion/main.go

```diff
@@ -0,0 +1,330 @@
+	}
+
+	if err := os.WriteFile(outputFile, outputData, 0644); err != nil {
+		return fmt.Errorf("failed to write output file: %w", err)
+	}
+
+	return nil
+}
+
+func createPayload(resourceType string, values map[string]interface{}, status string) map[string]interface{} {
+	if createFunc, exists := convert_funcs.ResourceMap[resourceType]; exists {
+		payload := createFunc(values, status)
+		return payload
+	}
+	return nil
+}
+
+func createPayloadList(plan Plan) []map[string]interface{} {
+	var data []map[string]interface{}
+
+	for _, change := range plan.Changes {
+		if len(change.Change.Actions) > 0 && change.Change.Actions[0] == "delete" {
+			payload := createPayload(change.Type, change.Change.Before, "deleted")
+			if payload != nil {
+				data = append(data, payload)
+			}
+		}
+	}
+
+	for _, resource := range plan.PlannedValues.RootModule.Resources {
+		payload := createPayload(resource.Type, resource.Values, "")
+		if payload != nil {
+			for _, value := range payload {
+				if obj, ok := value.(map[string]interface{}); ok {
+					if attributes, ok := obj["attributes"].(map[string]interface{}); ok {
+						if parentDn, ok := resource.Values["parent_dn"].(string); ok && parentDn != "" {
+							attributes["parent_dn"] = parentDn
+						}
+					}
+				}
+			}
+			data = append(data, payload)
+		}
+	}
+
+	return data
+}
+
+type TreeNode struct {
+	Attributes map[string]interface{} `json:"attributes,omitempty"`
+	Children   map[string]*TreeNode   `json:"children,omitempty"`
+	ClassName  string                 `json:"-"`
+}
+
+func NewTreeNode(className string, attributes map[string]interface{}) *TreeNode {
+	return &TreeNode{
+		ClassName:  className,
+		Attributes: attributes,
+		Children:   make(map[string]*TreeNode),
+	}
+}
+
+func constructTree(resources []map[string]interface{}) map[string]interface{} {
+	rootNode := NewTreeNode("uni", map[string]interface{}{"dn": "uni"})
+
+	dnMap := make(map[string]*TreeNode)
+	dnMap["uni"] = rootNode
+
+	for _, resourceList := range resources {
+		for resourceType, resourceData := range resourceList {
+			resourceAttributes := resourceData.(map[string]interface{})
+			attributes := safeMapInterface(resourceAttributes, "attributes")
+			dn := safeString(attributes, "dn")
+
+			var children []map[string]interface{}
+			if rawChildren, ok := resourceAttributes["children"].([]interface{}); ok {
+				for _, child := range rawChildren {
+					if childMap, ok := child.(map[string]interface{}); ok {
+						children = append(children, childMap)
+					}
+				}
+			}
+
+			buildParentPath(dnMap, rootNode, resourceType, dn, attributes, children)
+		}
+	}
+
+	return map[string]interface{}{rootNode.ClassName: exportTree(rootNode)}
+}
+
+func buildParentPath(dnMap map[string]*TreeNode, rootNode *TreeNode, resourceType, dn string, attributes map[string]interface{}, children []map[string]interface{}) {
+	if dn == "" && resourceType == "" {
+		return
+	}
+
+	cursor := rootNode
+	if dn != "" {
```

Comment:
this conditional seems redundant when above you exit the function already when it is ""